server/src/domain/person/person.service.ts

```diff
@@ -442,46 +443,59 @@ export class PersonService {
     return true;
   }
 
-  async mergePerson(auth: AuthDto, id: string, dto: MergePersonDto): Promise<BulkIdResponseDto[]> {
+  async mergePerson(auth: AuthDto, id: string, dto: MergePersonDto): Promise<MergePersonResponseDto> {
     const mergeIds = dto.ids;
+    if (!(mergeIds instanceof Array)) {
+      throw new BadRequestException('ids is not an array');
+    }
     await this.access.requirePermission(auth, Permission.PERSON_WRITE, id);
-    const primaryPerson = await this.findOrFail(id);
+    let primaryPerson = await this.findOrFail(id);
     const primaryName = primaryPerson.name || primaryPerson.id;
 
     const results: BulkIdResponseDto[] = [];
 
     const allowedIds = await this.access.checkAccess(auth, Permission.PERSON_MERGE, mergeIds);
 
-    for (const mergeId of mergeIds) {
-      const hasAccess = allowedIds.has(mergeId);
+    for (let i = 0; i < mergeIds.length; i++) {
+      const hasAccess = allowedIds.has(mergeIds[i]);
       if (!hasAccess) {
-        results.push({ id: mergeId, success: false, error: BulkIdErrorReason.NO_PERMISSION });
+        results.push({ id: mergeIds[i], success: false, error: BulkIdErrorReason.NO_PERMISSION });
         continue;
       }
 
       try {
-        const mergePerson = await this.repository.getById(mergeId);
+        const mergePerson = await this.repository.getById(mergeIds[i]);
         if (!mergePerson) {
-          results.push({ id: mergeId, success: false, error: BulkIdErrorReason.NOT_FOUND });
+          results.push({ id: mergeIds[i], success: false, error: BulkIdErrorReason.NOT_FOUND });
           continue;
         }
-
+        if (mergeIds.length === 1 && i === 0) {
```

Comment:
why does it only work for the first person? Why can't, inside the for loop, you just have logic like "if person doesn't have a name or or birth day and the current merge target does, save it". I don't see why we need specialized logic that only applies for the first person.

Reply:
I think this is a highly specialized case and it is a bit weird to figure out when it will copy data and when it won't. Can we not just have something generic like "when merging person, if the birthdate or name is empty, it will use the first available one from the merge candidates"?

Reply:
Sounds good